packages/core/strapi/src/cli/commands/transfer/command.ts

```diff
@@ -45,14 +45,142 @@ const command = () => {
-              'Exactly one remote source (from) or destination (to) option must be provided'
-            )
+            exitWith(1, 'Only one remote source (from) or destination (to) option may be provided')
         )
       )
+      .hook('preAction', async (thisCommand) => {
+        const opts = thisCommand.opts();
+        const hasEnvUrl = process.env.STRAPI_TRANSFER_URL;
+        const hasEnvToken = process.env.STRAPI_TRANSFER_TOKEN;
+
+        const logDocumentation = () => {
+          console.info(
+            'ℹ️  Data transfer documentation: https://docs.strapi.io/dev-docs/data-management/transfer'
+          );
+        };
+
+        const logEnvironmentVariables = () => {
+          if (!hasEnvUrl && !hasEnvToken) {
+            console.info('ℹ️  No transfer configuration found in environment variables');
+            console.info(
+              '   → Add STRAPI_TRANSFER_URL and STRAPI_TRANSFER_TOKEN environment variables to make the transfer process faster for future runs'
+            );
+            return;
+          }
+
+          console.info('ℹ️  Found transfer configuration in your environment:');
+
+          if (hasEnvUrl) {
+            console.info(
+              `   → Environment STRAPI_TRANSFER_URL (${hasEnvUrl}) will be used as the transfer URL`
+            );
+          }
+
+          if (hasEnvToken) {
+            console.info(
+              '   → Environment STRAPI_TRANSFER_TOKEN value will be used as the transfer token'
+            );
+          }
+
+          console.info(); // Empty line for better readability
+        };
+
+        const determineDirection = async () => {
+          // If user has not provided a direction from CLI, log the documentation
+          if (!opts.from && !opts.to) {
+            logDocumentation();
+          }
+
+          logEnvironmentVariables();
+
+          if (opts.from) {
+            return opts.from;
+          }
+          if (opts.to) {
+            return opts.to;
+          }
+
+          const { dir } = await inquirer.prompt([
+            {
+              type: 'list',
+              name: 'dir',
+              message: 'Choose transfer direction:',
+              choices: [
+                { name: 'Pull data from remote Strapi to local', value: 'from' },
+                { name: 'Push local data to remote Strapi', value: 'to' },
+              ],
+            },
+          ]);
+
+          return dir;
+        };
+
+        const determineUrl = async (direction: 'from' | 'to') => {
+          if (opts[direction]) {
+            return new URL(opts[direction]);
+          }
+
+          if (process.env.STRAPI_TRANSFER_URL) {
+            return new URL(process.env.STRAPI_TRANSFER_URL);
+          }
+
+          const answer = await inquirer.prompt([
+            {
+              type: 'input',
+              name: 'remoteUrl',
+              message: `Enter the URL of the remote Strapi instance to ${direction === 'from' ? 'get data from' : 'send data to'}:`,
```

Comment:
Should we make it clear from the beginning that the /admin part of the path is expected? (currently it's only in the error handling)

Reply:
@maccomaccomaccomacco what do you think? I think it's a good idea I just don't see how to reasonably fit it into one line.

Reply:
I don't think we can do much more than say "Enter the _admin_ URL of the remote Strapi instance to..."

Reply:
For now let's leave it, but I think a future improvement could be that if we do not detect it ends with '/admin' we log a warning message. That would basically solve the problem and we wouldn't need the more complex solutions we've discussed before (like appending it and retrying connection)

Reply:
It's not always /admin as that can be changed so we need to accommodate for that if we do

Reply:
Can't we just let the user put the base Url and get the admin path from the admin configuration?